tests/benchmarks/test_xgboost.py

```diff
@@ -0,0 +1,237 @@
+    upper_bound = Q3 + (1.5 * (Q3 - lower_bound))
+    ddf = ddf.loc[(ddf["trip_time"] >= lower_bound) & (ddf["trip_time"] <= upper_bound)]
+
+    ###################
+    # Join with domains
+    ###################
+    ddf = dd.merge(
+        ddf,
+        taxi_zone_lookup,
+        how="inner",
+        left_on="PULocationID",
+        right_on="LocationID",
+    )
+    ddf = ddf.rename(columns={"Borough": "PUBorough", "Superborough": "PUSuperborough"})
+    ddf = ddf.drop(columns="LocationID")
+
+    ddf = dd.merge(
+        ddf,
+        taxi_zone_lookup,
+        how="inner",
+        left_on="DOLocationID",
+        right_on="LocationID",
+    )
+    ddf = ddf.rename(columns={"Borough": "DOBorough", "Superborough": "DOSuperborough"})
+    ddf = ddf.drop(columns="LocationID")
+
+    ddf["PUSuperborough_DOSuperborough"] = ddf.PUSuperborough.str.cat(
+        ddf.DOSuperborough, sep="-"
+    )
+    ddf = ddf.drop(columns=["PUSuperborough", "DOSuperborough"])
+
+    ############
+    # Categorize
+    ############
+    categories = [
+        "hvfhs_license_num",
+        "PULocationID",
+        "DOLocationID",
+        "wav_request_flag",
+        "accessible_vehicle",
+        "pickup_month",
+        "pickup_dow",
+        "pickup_hour",
+        "PUBorough",
+        "DOBorough",
+        "PUSuperborough_DOSuperborough",
+    ]
+
+    # TODO https://github.com/dask/dask/issues/9847
+    ddf = ddf.astype(dict.fromkeys(categories, "category"))
+    ddf = ddf.persist()
+    # This blocks until the whole workload so far has been persisted
+    ddf = ddf.categorize(categories)
+
+    ########
+    # Output
+    ########
+    ddf = ddf.persist().repartition(partition_size="100MB")
+    # At this point we would normally finish with to_parquet()
+    wait(ddf, small_client, timeout=600)
+
+
+@run_up_to_nthreads("small_cluster", 200, reason="fixed size dataset")
+def test_optuna_hpo(small_client):
+    # Needs either coiled-runtime >0.2.1 or explicitly adding the package to the A/B
+    # test environment
+    optuna = pytest.importorskip("optuna")
+
+    #############################
+    # Dataset load and preprocess
+    #############################
+    # Load feature table generated by previous test
+    ddf = dd.read_parquet(
+        "s3://coiled-runtime-ci/xgboost/nyc-uber-lyft-preprocessed.parquet",
+    )
+
+    # Full size dataset on 40x r6i.large
+    ddf = downscale_dataframe(ddf, "580 GiB")
+
+    # Under the hood, XGBoost converts floats to `float32`.
+    # Let's do it only once here.
+    float_cols = ddf.select_dtypes(include="float").columns.tolist()
+    ddf = ddf.astype({c: np.float32 for c in float_cols})
+
+    # We need the categories to be known
+    categorical_vars = ddf.select_dtypes(include="category").columns.tolist()
+
+    # categorize() reads the whole input and then discards it.
+    # Let's read from disk only once.
+    ddf = ddf.persist()
+    ddf = ddf.categorize(columns=categorical_vars)
```

Comment:
If the columns are already a category, is this line doing anything?
ddf.categorize converts into categories the columns, but they already are categories. https://docs.dask.org/en/stable/generated/dask.dataframe.DataFrame.categorize.html

Reply:
On a to_parquet / read_parquet round-trip, you lose the domains knowledge in the metadata, which the original notebook states xgboost needs to know (I did not investigate the veridicity of this statement).

Calling categorize() again restores the domain info in the metadata, albeit in an unnecessarily expensive way (you could just read them from the first partition).